src/main/resources/requests/SubscriptionRequests.ts

```diff
@@ -92,4 +92,14 @@ export class SubscriptionRequests {
         }
         return null;
     }
+
+    public async fulfillSubscriptions(artefact): Promise<string> {
```

Comment:
I'm thinking we want to add an audit log for the submission action.

This is slightly out of scope of this ticket, however I do think it is needed before merging so would be good to discuss where we add this.